test/people.js

```diff
@@ -1,259 +1,242 @@
 const Mixpanel = require('../lib/mixpanel-node');
-const Sinon = require('sinon');
 const {create_profile_helpers} = require('../lib/profile_helpers');
 
-// shared test case
-const test_send_request_args = function(test, func, {args, expected, use_modifiers, use_callback} = {}) {
-    let expected_data = {$token: this.token, $distinct_id: this.distinct_id, ...expected};
-    let callback;
-
-    args = [this.distinct_id, ...(args ? args : [])];
-
-    if (use_modifiers) {
-        var modifiers = {
-            '$ignore_alias': true,
-            '$ignore_time': true,
-            '$ip': '1.2.3.4',
-            '$time': 1234567890,
-            '$latitude': 40.7127753,
-            '$longitude': -74.0059728,
-        };
-        Object.assign(expected_data, modifiers);
-        args.push(modifiers);
-    }
-    if (use_callback) {
-        callback = function() {};
-        args.push(callback);
-    }
-
-    this.mixpanel.people[func](...args);
-
-    const expectedSendRequestArgs = [{ method: 'GET', endpoint: this.endpoint, data: expected_data }];
-    test.ok(
-        this.send_request.calledWithMatch(...expectedSendRequestArgs),
-        `people.${func} didn't call send_request with correct arguments
-        Actual arguments:    ${JSON.stringify(this.send_request.getCall(0).args)}
-        expected to include: ${JSON.stringify(expectedSendRequestArgs)}`
-    );
-    if (use_callback) {
-        test.ok(
-            this.send_request.args[0][1] === callback,
-            `people.${func} didn't call send_request with a callback`
-        );
-    }
-    test.done();
-};
-
-exports.people = {
-    setUp: function(next) {
-        this.endpoint = '/engage';
-        this.distinct_id = 'user1';
-        this.token = 'token';
-
-        this.send_request = Sinon.stub();
-
-        this.mixpanel = Mixpanel.init(this.token);
-        this.mixpanel.send_request = this.send_request
-
-        this.test_send_request_args = test_send_request_args;
-
-        next();
-    },
-
-    tearDown: function(next) {
-        next();
-    },
-
-    _set: {
-        "handles set_once correctly": function(test){
-            this.test_send_request_args(test, 'set_once', {
+describe('people', () => {
+    const endpoint = '/engage';
+    const distinct_id = 'user1';
+    const token = 'token';
+    let send_request;
+    let mixpanel;
+    beforeEach(() => {
+        send_request = vi.fn();
+
+        mixpanel = Mixpanel.init(token);
+        mixpanel.send_request = send_request
```

Comment:
Seems like you're inconsistent between using this approach and `vi.spyOn(mixpanel, 'send_request')`. I would think spyOn is preferable, but is there some context I'm missing?

Reply:
good catch.. i think when i did the conversion, this was the way sinon stubs this function.